rest-client/src/main/java/io/apicurio/registry/client/request/ResultCallback.java

```diff
@@ -71,7 +92,18 @@ public void onResponse(Call<T> call, retrofit2.Response<T> response) {
 
     @Override
     public void onFailure(Call<T> call, Throwable t) {
-        result.completeExceptionally(t);
+        Throwable rootCause = extractRootCause(t);
+        if (rootCause instanceof HttpResponseException) {
+            //authorization error
```

Comment:
This will likely be the case but maybe is a good idea to also check the error code since Keycloak can throw other exceptions, not just the 401.